File: java/Problem055_JumpGame.java
// https://leetcode.com/problems/jump-game

class Solution {
    public boolean canJump(int[] nums) {
        int size = nums.length-1;
        int first = nums[0];

        if (first > size) return false;

        int jump = nums[first];

        if (jump > size) return false;

        int result = nums[jump]+jump;

        return (result == size);
    }
}

File: go/problem055_jump_game.go
// https://leetcode.com/problems/jump-game

func canJump(nums []int) bool {
    maxJump := 0

    for i:=0; i<len(nums); i++ {
        if (i>maxJump) {
            return false
        }
        maxJump = max(maxJump, i+nums[i])
    }
    return true
}
